Dockerfile

```diff
@@ -1,14 +1,40 @@
-FROM node:20-slim
+FROM node:20-slim as builder
 
 RUN apt-get update
-RUN apt-get install -y build-essential cmake git libssl-dev
+RUN apt-get install -y build-essential cmake git libssl-dev tini
 
 WORKDIR /app
 
-COPY . .
+COPY package*.json ./
+
 RUN npm ci --quiet
+
+COPY . .
+
 RUN npm run build
 
+RUN npm prune --omit=dev
+
+FROM node:20-slim as app
+ENV NODE_ENV production
+WORKDIR /app
+# built src without dev dependencies
+COPY --from=builder /app ./
+# tini is used to handle signals properly, see https://github.com/krallin/tini#using-tini
+COPY --from=builder /usr/bin/tini /usr/bin/tini
+
+# copy shared libraries (without having artifacts from apt-get install that is needed to build our application)
+COPY --from=builder /usr/lib/**/libcrypto* /usr/lib/
+COPY --from=builder /usr/lib/**/libssl* /usr/lib/
```

Comment:
we can re-install libssl and it would re-use the cached image, but that was more troublesome due to some additional things being installed by other apt-get installed packges: `build-essential cmake git libssl-dev` and I didn't want to reinstall all of them.